spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/CredhubEnvironmentRepository.java

```diff
@@ -0,0 +1,63 @@
+package org.springframework.cloud.config.server.environment;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.cloud.config.environment.Environment;
+import org.springframework.cloud.config.environment.PropertySource;
+import org.springframework.credhub.core.CredHubOperations;
+import org.springframework.credhub.support.CredentialDetails;
+import org.springframework.credhub.support.SimpleCredentialName;
+import org.springframework.credhub.support.json.JsonCredential;
+import org.springframework.util.StringUtils;
+
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * @author Alberto C. Ríos
+ */
+public class CredhubEnvironmentRepository implements EnvironmentRepository {
+
+	private CredHubOperations credHubOperations;
+
+	public CredhubEnvironmentRepository(CredHubOperations credHubOperations) {
+		this.credHubOperations = credHubOperations;
+	}
+
+	@Override
+	public Environment findOne(String application, String profilesList, String label) {
+		if (StringUtils.isEmpty(profilesList)) {
+			profilesList = "default";
+		}
+		if (StringUtils.isEmpty(label)) {
+			label = "master";
```

Comment:
Does master make sense here?

Reply:
I don't think it does make a lot of sense, however, a lot of the repositories, including `Jdbc` are using `master` as a default label.
I am happy to remove the default and just do not concatenate any label if nothing was specified, but for the sake of consistency, I decided to add that there. What do you think?